### PR TITLE
fix races in http cors

### PR DESCRIPTION
**Pull request by miolini**
License: MIT
Signed-off-by: Artem Andreenko <mio@volmy.com>

### CODE REVIEW

**Comment by miolini** [comment]
@jbenet I guess travis failed on different problem.

**Comment by jbenet** [comment]
Thanks @miolini -- LGTM @whyrusleeping  ?

**Comment by whyrusleeping** [comment]
Yeap, LGTM.

File: commands/http/handler.go [hunk 1]
```diff
@@ -333,8 +382,8 @@ func allowOrigin(r *http.Request, cfg *ServerConfig) bool {
 	if origin == "" {
 		return true
 	}
-
-	for _, o := range cfg.CORSOpts.AllowedOrigins {
+	origins := cfg.AllowedOrigins()
+	for _, o := range origins {
```
Comment (whyrusleeping) [inline commands/http/handler.go]: could be written as:
```
for _,o := range cfg.AllowedOrigins() {
```

But not a big deal, i'm fine merging without.

**Pull request merged**